### PR TITLE
Fullscreen button

### PR DESCRIPTION
**Pull request by glewe**
I added a full screen button to the top menu as it was available in the BS4 version.
I saw that there is a PR for this already but it seems to be outdated.
(This is my first PR here so hopefully I did it right).
Best regards,
George

### CODE REVIEW

File: src/ts/fullscreen.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+/**
+ * Class Definition.
+ * ============================================================================
+ */
+class FullScreen {
+  /**
+   * toggleFullScreen.
+   */
+  toggleFullScreen(): void {
+    const eleRequestFullscreen = document.getElementById(ID_REQUEST_FULLSCREEN_ICON)
```
Comment (danny007in) [inline src/ts/fullscreen.ts:29]: use querySelector

Comment (glewe) [reply src/ts/fullscreen.ts]: Fixed in lates commit

File: src/html/components/dashboard/_topbar.astro [hunk 2]
```diff
@@ -154,6 +154,15 @@ const distPath = path;
       </li>
       <!--end::Notifications Dropdown Menu-->
 
+      <!--begin::Fullscreen Toggle-->
+      <li class="nav-item">
+        <a class="nav-link" href="#" data-lte-toggle="fullscreen">
+          <i id="btn-request-fullscreen" class="bi bi-arrows-fullscreen"></i>
```
[the remark this answers is not in the document]
Comment (glewe) [reply src/html/components/dashboard/_topbar.astro]: Fixed in latest commit

File: src/ts/fullscreen.ts [hunk 3]
```diff
@@ -14,8 +14,8 @@ import {
  * ============================================================================
  */
 const SELECTOR_FULLSCREEN_TOGGLE = '[data-lte-toggle="fullscreen"]'
-const ID_REQUEST_FULLSCREEN_ICON = 'btn-request-fullscreen'
-const ID_EXIT_FULLSCREEN_ICON = 'btn-exit-fullscreen'
+const ID_REQUEST_FULLSCREEN_ICON = '[data-lte-icon="request-fullscreen"]'
```
Comment (danny007in) [inline src/ts/fullscreen.ts:17]: rename `ID_REQUEST_FULLSCREEN_ICON`

File: src/html/components/dashboard/_topbar.astro [hunk 4]
```diff
@@ -157,8 +157,8 @@ const distPath = path;
       <!--begin::Fullscreen Toggle-->
       <li class="nav-item">
         <a class="nav-link" href="#" data-lte-toggle="fullscreen">
-          <i id="btn-request-fullscreen" class="bi bi-arrows-fullscreen"></i>
-          <i id="btn-exit-fullscreen" class="bi bi-fullscreen-exit" style="display: none;"></i>
+          <i data-lte-icon="request-fullscreen" class="bi bi-arrows-fullscreen"></i>
```
Comment (danny007in) [inline src/html/components/dashboard/_topbar.astro:160]: rename as https://github.com/ColorlibHQ/AdminLTE/blob/0cb0400d8ddecb5e2ac74f9660a28de6dae16687/src/html/pages/widgets/cards.astro#L143
like `maximum`

**Review by danny007in — CHANGES_REQUESTED** [review]

File: src/ts/fullscreen.ts (hunk 3 above)
Comment (glewe) [reply src/ts/fullscreen.ts]: Makes sense. Renamed in lates commit.

File: src/html/components/dashboard/_topbar.astro (hunk 4 above)
Comment (glewe) [reply src/html/components/dashboard/_topbar.astro]: Makes sense. Renamed in lates commit.

File: package-lock.json
Comment (danny007in) [inline package-lock.json:1]: remove package file, and i will merge this PR

**Review by danny007in — CHANGES_REQUESTED** [review]

File: package-lock.json
Comment (glewe) [reply package-lock.json:1]: I rolled back the changes of package-lock.json to the master branch.

File: src/ts/fullscreen.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+} from './util/index'
+
+/**
+ * Constants
+ * ============================================================================
+ */
+const SELECTOR_FULLSCREEN_TOGGLE = '[data-lte-toggle="fullscreen"]'
+const SELECTOR_MAXIMIZE_ICON = '[data-lte-icon="maximize"]'
+const SELECTOR_MINIMIZE_ICON = '[data-lte-icon="minimize"]'
+
```
Comment (danny007in) [inline src/ts/fullscreen.ts:19]: Add Event and dispatch

**Review by danny007in — CHANGES_REQUESTED** [review]

File: src/ts/fullscreen.ts (hunk 5 above)
Comment (glewe) [reply src/ts/fullscreen.ts:19]: Done in latest commit

**Comment by danny007in** [comment]
on `esc` key press, icon is not changing

https://github.com/ColorlibHQ/AdminLTE/assets/50356015/7d391b17-efca-4538-84b6-f1a92bea9a96

**Comment by danny007in** [comment]
Make PR commit readable https://www.youtube.com/watch?v=gXCkYkLQ3To

**Comment by glewe** [comment]
> on esc key press, icon is not changing
I added an keydown event listener to the fullscreen.ts class. However, I could not get it to change the icon on the first Esc. It takes two Esc events to change. Can you have a look?

**Comment by danny007in** [comment]
@glewe 
I'm going to make some changes to the code. Please don't push anything

**Comment by glewe** [comment]
> @glewe 
> I'm going to make some changes to the code. Please don't push anything

Ok. Thanks.

**Comment by danny007in** [comment]
`f11` and `esc` are not stable with fullscreenmode.

once check it please. i pushed some updates

**Pull request merged**